kyaml/setters2/set_test.go

```diff
@@ -1395,6 +1395,35 @@ func TestValidateAgainstSchema(t *testing.T) {
 			shouldValidate:   false,
 			expectedErrorMsg: "foo in body must be of type integer",
 		},
+		{
+			name: "all values should satisfy type string",
```

Comment:
Can we add a case where there are 2 validations by Charlie which are, `type: string, maxLength:3`, Alice sets value to 1234, and how the error looks like ?

Reply:
Done